azure-pipelines.yml

```diff
@@ -1,38 +1,42 @@
 trigger:
-- master
+- updates/strategy-tweak
 
-jobs:
-- job: Build_Linux_based_images
-  displayName: Build Linux based images
-  pool:
-    vmImage: 'ubuntu-16.04'
-  steps:
-  - template: yml-steps/build-and-push.yml
-    parameters:
-      targetStage: 'linux-arm'
-      repository: 'myapp'
-      dockerfile: 'Dockerfile.Linux'
-  - template: yml-steps/build-and-push.yml
-    parameters:
-      targetStage: 'linux-x64'
-      repository: 'myapp'
-      dockerfile: 'Dockerfile.Linux'
+#required - add in the below variables
+variables:
+  someKey: someValue
+#  REGISTRYURL: <insert name here>.azurecr.io
+#  REGISTRY_USERNAME: ??
+#  REGISTRY_PASSWORD: ??  #use secret here!
 
-- job: Build_Windows_based_images
-  displayName: Build Windows based images
+jobs:
+- job: Build_images
+  displayName: Build images
+  strategy:
+    matrix:
+      Linux:
+        vmImage: 'ubuntu-16.04'
+        imageName: 'linux-arm'
+        dockerfile: 'Dockerfile.Linux'
+      MacOS:
+        vmImage: 'ubuntu-16.04'
+        imageName: 'linux-x64'
+        dockerfile: 'Dockerfile.Linux'
+      Windows:
+        vmImage: 'windows-latest'
+        imageName: 'windows-x64'
+        dockerfile: 'Dockerfile.Windows'
   pool:
-    vmImage: 'windows-2019'
+    vmImage: $(vmImage)
   steps:
   - template: yml-steps/build-and-push.yml
     parameters:
-      targetStage: 'windows-x64'
+      targetStage: $(imageName)
       repository: 'myapp'
-      dockerfile: 'Dockerfile.Windows'
+      dockerfile: $(dockerfile)
 
```

Comment:
This matrix thing is great... actually, it feels like one of the main scenarios for this feature. Many thanks for the hint!